plugins/catalyst/lib/Conversion/CatalystQuantumToMQTOpt/CatalystQuantumToMQTOpt.cpp

```diff
@@ -0,0 +1,421 @@
+/*
+ * Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+ * Copyright (c) 2025 Munich Quantum Software Company GmbH
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Licensed under the MIT License
+ */
+
+#include "mlir/Conversion/CatalystQuantumToMQTOpt/CatalystQuantumToMQTOpt.h"
```

Comment:
We still need to do something about clang-tidy not working correctly for the plugin.
The main reason for that is that the plugin build is completely loose from the main mqt-core build. Hence, there is no compilation database, and clang-tidy has problems finding the headers in the build directory.

As a first step, we should probably figure out how to best ignore the `plugins` directory from the clang-tidy run. However, mid-term, I would really want to have a clang-tidy check running properly over the code here. CLion locally revealed dozens of warnings that were fixed in the last couple of commits.